xapian-core/weight/lmweight.cc

```diff
@@ -200,16 +222,34 @@ LMWeight::get_maxpart() const
     return (upper_bound * param_log > 1.0) ? log(upper_bound * param_log) : 1.0;
 }
 
+
+/* The extra weight component in the Dir+ formula is :-
+ *
+ * |Q| * log (param_smoothing1 / (|D| + param_smoothing1))
+ * 
+ * where, |Q| is total query length.
+ *	  |D| is total document length.
+ */
 double
-LMWeight::get_sumextra(Xapian::termcount, Xapian::termcount) const
+LMWeight::get_sumextra(Xapian::termcount len, Xapian::termcount) const
 {
-    return 0;
+    if (select_smoothing == DIRICHLET_PLUS_SMOOTHING) {
+	double extra_weight = param_smoothing1 / (len + param_smoothing1);
+	return get_query_length() * log(extra_weight);
+    }
+    else
+	return 0;
 }
 
 double
 LMWeight::get_maxextra() const
 {
-    return 0;
+    if (select_smoothing == DIRICHLET_PLUS_SMOOTHING) {
+	double extra_weight = param_smoothing1 / (get_doclength_lower_bound() + param_smoothing1);
```

Comment:
@ojwb Test case `unigramlmweight7` is failing under valgrind with error message telling that an uninitialised variable is being used at line 248 :

```bash
Running test: unigramlmweight7...==22742== Conditional jump or move depends on uninitialised value(s)
==22742==    at 0x55E0D88: log (w_log.c:28)
==22742==    by 0x4FC6543: Xapian::LMWeight::get_maxextra() const (lmweight.cc:248)
```
I might be missing something here but I don't see any uninitialised variable in this line. Any pointers to help fix this failing test?